packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -110,27 +143,30 @@ export class CanvasManager implements CanvasManagerInterface {
     } = options;
     this.mutationCb = options.mutationCb;
     this.mirror = options.mirror;
-
-    callbackWrapper(() => {
-      if (recordCanvas && sampling === 'all')
-        this.initCanvasMutationObserver(
-          win,
-          blockClass,
-          blockSelector,
-          unblockSelector,
-        );
-      if (recordCanvas && typeof sampling === 'number')
-        this.initCanvasFPSObserver(
-          sampling,
-          win,
-          blockClass,
-          blockSelector,
-          unblockSelector,
-          {
-            dataURLOptions,
-          },
-        );
-    })();
+    this.window = options.win;
+
+    if (!options.manualSnapshot) {
```

Comment:
You could invert this and do an early return, this makes it a bit easier to read (less nesting).

```suggestion
    if (options.manualSnapshot) {
      return;
    }
```